src/Databases/DatabaseReplicated.cpp

```diff
@@ -962,7 +962,7 @@ void DatabaseReplicated::recoverLostReplica(const ZooKeeperPtr & current_zookeep
         /// And QualifiedTableName::parseFromString doesn't handle this.
         auto qualified_name = QualifiedTableName{.database = getDatabaseName(), .table = table_name};
         auto query_ast = parseQueryFromMetadataInZooKeeper(table_name, create_table_query);
-        tables_dependencies.addDependencies(qualified_name, getDependenciesFromCreateQuery(getContext(), qualified_name, query_ast));
+        tables_dependencies.addDependencies(qualified_name, getLoadingDependenciesFromCreateQuery(getContext(), qualified_name, query_ast));
```

Comment:
But we do need "create dependencies" here, like in backups. The problem is that "create dependencies" include some "false-positive" dependencies: we don't need `default.raw_data` to create `my_buffer` (see the example in https://github.com/ClickHouse/ClickHouse/issues/50222). You will get the same error about "cyclic dependencies" if you run create queries from https://github.com/ClickHouse/ClickHouse/issues/50222, make a backup, and then try to restore